Assets/MixedRealityToolkit.Services/InputSystem/NearInteractionTouchableUnityUI.cs

```diff
@@ -17,7 +15,7 @@ public class NearInteractionTouchableUnityUI : BaseNearInteractionTouchable
     {
         private RectTransform rectTransform;
 
-        public static IReadOnlyCollection<NearInteractionTouchableUnityUI> Instances { get { return instances.AsReadOnly(); } }
+        public static List<NearInteractionTouchableUnityUI> Instances => instances;
```

Comment:
I don't think this retains the original intent, which was to prevent the collection from being added to / removed from.

Reply:
Is there a way to deliver a read-only collection without garbage / that you can iterate over without an enumerator? I feel like there must be.

Reply:
I think that List itself also just implements IReadOnlyCollection:

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1?view=netframework-4.8

So you might be able to just return instances itself (and leave the IReadOnlyCollection part alone)

Reply:
Tried changing Instances to return an IReadOnlyList and it works great.